### PR TITLE
copy 1.2 upgrade to 1.3 and rename path to 1.3

### PR DESCRIPTION
**Pull request by dzou-hpe**
## Summary and Scope

_Summarize what has changed. Explain why this PR is necessary. What is impacted? Is this a new feature, critical bug fix, etc?_

_Is this change backwards incompatible, backwards compatible, or a backwards compatible bugfix?_

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [issue id](issue link)
* Change will also be needed in `<insert branch name here>`
* Future work required by [issue id](issue link)
* Documentation changes required in [issue id](issue link)
* Merge with/before/after `<insert PR URL here>`

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * `<development system>`
  * Local development environment
  * Virtual Shasta

### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: upgrade/1.3/Stage_0_Prerequisites.md [hunk 1]
```diff
@@ -176,12 +176,12 @@ If the following command does not complete successfully, check if the `TOKEN` en
...
 ## Stage 0.3 - Upgrade management network
 
-### Verify that switches have 1.2 configuration in place
+### Verify that switches have 1.3 configuration in place
 
 1. Log in to each management switch.
 
    ```bash
-   linux# ssh admin@1.2.3.4
+   linux# ssh admin@1.3.3.4
```
Comment (mharding-hpe) [inline upgrade/1.3/Stage_0_Prerequisites.md:184]:
```suggestion
   linux# ssh admin@1.2.3.4
```

File: upgrade/1.3/Stage_0_Prerequisites.md [hunk 2]
```diff
@@ -190,14 +190,14 @@ If the following command does not complete successfully, check if the `TOKEN` en
...
+   # CSM version:  1.3
    # CANU version: 1.3.2
    ##################################################################################
    ```
 
-   - Output like the above text means that the switches have a CANU-generated configuration for CSM 1.2 in place. In this case, follow the steps in
-     [Management Network 1.0 (`1.2 Preconfig`) to 1.2](../../operations/network/management_network/1.0_to_1.2_upgrade.md).
-   - If the banner does NOT contain text like the above, then contact support in order to get the `1.2 Preconfig` applied to the system.
+   - Output like the above text means that the switches have a CANU-generated configuration for CSM 1.3 in place. In this case, follow the steps in
+     [Management Network 1.2 (`1.3 Preconfig`) to 1.3](../../operations/network/management_network/1.0_to_1.3_upgrade.md).
```
Comment (mharding-hpe) [inline upgrade/1.3/Stage_0_Prerequisites.md:199]: Is that a valid link target?

Comment (mharding-hpe) [inline upgrade/1.3/Stage_0_Prerequisites.md:199]: Is there going to be a 1.3 Preconfig? I didn't think we were making the same big network changes moving to that release.

File: upgrade/1.3/scripts/sls/README.SLS_Upgrade.md [hunk 3]
```diff
@@ -48,15 +48,15 @@ During the update of SLS, at a minimum, answers to the following questions must
     * Example 1: Upgrade, using the CHN as the system default route (will by default output to `migrated_sls_file.json`).
 
         ```bash
-        ncn# ./sls_updater_csm_1.2.py --sls-input-file sls_input_file.json \
+        ncn# ./sls_updater_csm_1.3.py --sls-input-file sls_input_file.json \
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/README.SLS_Upgrade.md:51]: The actual script file is not being renamed with this PR, so this command won't exist

File: upgrade/1.3/scripts/sls/README.SLS_Upgrade.md [hunk 4]
```diff
@@ -48,15 +48,15 @@ During the update of SLS, at a minimum, answers to the following questions must
...
                 --bican-user-network-name CHN \
                 --customer-highspeed-network 5 10.103.11.192/26
         ```
 
     * Example 2: Upgrade, using the CAN as the system default route, keep the generated CHN (for testing), and preserve the existing external-dns entry.
 
         ```bash
-        ncn# ./sls_updater_csm_1.2.py --sls-input-file sls_input_file.json \
+        ncn# ./sls_updater_csm_1.3.py --sls-input-file sls_input_file.json \
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/README.SLS_Upgrade.md:59]: The actual script file is not being renamed with this PR, so this command won't exist

File: upgrade/1.3/scripts/sls/README.SLS_Upgrade.md [hunk 5]
```diff
@@ -77,7 +77,7 @@ During the update of SLS, at a minimum, answers to the following questions must
 For help and all options, run the following:
 
 ```bash
-ncn# ./sls_updater_csm_1.2.py --help
+ncn# ./sls_updater_csm_1.3.py --help
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/README.SLS_Upgrade.md:80]: The actual script file is not being renamed with this PR, so this command won't exist

File: upgrade/1.3/scripts/sls/README.SLS_Upgrade.md [hunk 6]
```diff
@@ -107,12 +107,12 @@ This needs to be done in the order listed above.
...
 ## Remove `api-gateway` / `istio-ingress-gateway` Reservations from HMNLB Subnets
 
-For CSM 1.2, the API gateway will no longer listen on the HMNLB MetalLB address pool.
+For CSM 1.3, the API gateway will no longer listen on the HMNLB MetalLB address pool.
 These aliases provided DNS records and are being removed.
 
 ## Create the BICAN Network "Toggle"
 
-New for CSM 1.2, the BICAN network `ExtraProperties` value of `SystemDefaultRoute` is used
+New for CSM 1.3, the BICAN network `ExtraProperties` value of `SystemDefaultRoute` is used
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/README.SLS_Upgrade.md:115]: That is not new for 1.3

File: upgrade/1.3/scripts/sls/README.SLS_Upgrade.md [hunk 7]
```diff
@@ -98,7 +98,7 @@ migration script is run and a new, migrated output file is created.
 
 ## Migrate Switch Names
 
-Switch names change in CSM 1.2 and must be applied in the following order:
+Switch names change in CSM 1.3 and must be applied in the following order:
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/README.SLS_Upgrade.md:101]: This is not a change for 1.3

File: upgrade/1.3/scripts/sls/README.SLS_Upgrade.md [hunk 8]
```diff
@@ -107,12 +107,12 @@ This needs to be done in the order listed above.
 
 ## Remove `api-gateway` / `istio-ingress-gateway` Reservations from HMNLB Subnets
 
-For CSM 1.2, the API gateway will no longer listen on the HMNLB MetalLB address pool.
+For CSM 1.3, the API gateway will no longer listen on the HMNLB MetalLB address pool.
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/README.SLS_Upgrade.md:110]: This is not a change for 1.3

File: upgrade/1.3/scripts/sls/csm_1_2_upgrade/__init__.py [hunk 9]
```diff
@@ -19,4 +19,4 @@
 # OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
 # ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 # OTHER DEALINGS IN THE SOFTWARE.
-"""Modifications to SLS to upgrade from CSM 1.0.x to CSM 1.2."""
+"""Modifications to SLS to upgrade from CSM 1.0.x to CSM 1.3."""
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/csm_1_2_upgrade/__init__.py:22]: I don't think you're allowed to upgrade from 1.0 to 1.3

File: upgrade/1.3/scripts/sls/csm_1_2_upgrade/sls_updates.py [hunk 10]
```diff
@@ -19,7 +19,7 @@
 # OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
 # ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 # OTHER DEALINGS IN THE SOFTWARE.
-"""Functions used to update SLS from CSM 1.0.x to CSM 1.2."""
+"""Functions used to update SLS from CSM 1.0.x to CSM 1.3."""
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/csm_1_2_upgrade/sls_updates.py:22]: I don't think you're allowed to upgrade from 1.0 to 1.3

File: upgrade/1.3/scripts/sls/csm_1_2_upgrade/sls_updates.py [hunk 11]
```diff
@@ -147,17 +147,17 @@ def sls_and_input_data_checks(
 
 
 def migrate_switch_names(networks, hardware):
-    """Rename CSM <=1.0.x switches to new CSM 1.2 naming.
+    """Rename CSM <=1.0.x switches to new CSM 1.3 naming.
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/csm_1_2_upgrade/sls_updates.py:150]: This shouldn't be something we'd do for a 1.3 upgrade

File: upgrade/1.3/scripts/sls/csm_1_2_upgrade/sls_updates.py [hunk 12]
```diff
@@ -147,17 +147,17 @@ def sls_and_input_data_checks(
...
 
-    CSM 1.2 changes switch names in the following order:
+    CSM 1.3 changes switch names in the following order:
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/csm_1_2_upgrade/sls_updates.py:152]: This is not new for 1.3

File: upgrade/1.3/scripts/sls/csm_1_2_upgrade/sls_updates.py [hunk 13]
```diff
@@ -147,17 +147,17 @@ def sls_and_input_data_checks(
...
       1. sw-leaf-xyz -> sw-leaf-bmc-xyz
       2. sw-agg-xyz -> sw-leaf-xyz
 
     Args:
         networks (sls_utils.Managers.NetworkManager): Dictionary of SLS networks
         hardware (json): Unmanaged SLS Hardware structure in JSON
     """
-    # The BICAN network only exists in CSM >=1.2.  If it exists do NOT migrate
+    # The BICAN network only exists in CSM >=1.3.  If it exists do NOT migrate
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/csm_1_2_upgrade/sls_updates.py:160]:
```suggestion
    # The BICAN network only exists in CSM >=1.2.  If it exists do NOT migrate
```

File: upgrade/1.3/scripts/sls/expert_mode.md [hunk 14]
```diff
@@ -26,18 +26,18 @@ The upgrader requires only two inputs:
 An example of this would be:
 
 ```bash
-ncn# ./sls_updater_csm_1.2.py \
+ncn# ./sls_updater_csm_1.3.py \
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/expert_mode.md:29]: Script needs to be renamed if this doc is using the new name

File: upgrade/1.3/scripts/sls/expert_mode.md [hunk 15]
```diff
@@ -26,18 +26,18 @@ The upgrader requires only two inputs:
...
 
-The above example will use *default values* for all other input values. A list of input parameters and default values can be found by running `./sls_updater_csm_1.2.py --help`.
+The above example will use *default values* for all other input values. A list of input parameters and default values can be found by running `./sls_updater_csm_1.3.py --help`.
 Likely using default VLAN and Network values will not be what is desired: The CAN or CHN are usually site-routable values.
 
 In this case an example minimal usable input while using CAN, would be:
 
 ```bash
-ncn# ./sls_updater_csm_1.2.py \
+ncn# ./sls_updater_csm_1.3.py \
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/expert_mode.md:40]: Script needs to be renamed if this doc is using the new name

File: upgrade/1.3/scripts/sls/expert_mode.md [hunk 16]
```diff
@@ -46,15 +46,15 @@ ncn# ./sls_updater_csm_1.2.py \
 For CHN the analog minimal usable input would be:
 
 ```bash
-ncn# ./sls_updater_csm_1.2.py \
+ncn# ./sls_updater_csm_1.3.py \
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/expert_mode.md:49]: Script needs to be renamed if this doc is using the new name

File: upgrade/1.3/scripts/sls/expert_mode.md [hunk 17]
```diff
@@ -72,22 +72,22 @@ Note that `external-dns` preservation is mutually exclusive from `ncns`. This is
 A very common (and recommended) next step minimal command line is as follows for a system desiring the CHN with no change of the `external-dns` value:
 
 ```bash
-ncn# ./sls_updater_csm_1.2.py \
+ncn# ./sls_updater_csm_1.3.py \
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/expert_mode.md:75]: Script not renamed to match doc change

File: upgrade/1.3/scripts/sls/expert_mode.md [hunk 18]
```diff
@@ -72,22 +72,22 @@ Note that `external-dns` preservation is mutually exclusive from `ncns`. This is
...
-existing CAN/CMN addresses on manager, worker, and storage NCNs during the upgrade to CSM 1.2.
+existing CAN/CMN addresses on manager, worker, and storage NCNs during the upgrade to CSM 1.3.
 
 **NOTE:** The output of the upgrader is to a file, not the screen. The logged output of the upgrader should be used as a guide and reviewed.
 
 For example, using:
 
 ```bash
-ncn# ./sls_updater_csm_1.2.py --sls-input-file sls_input_file.json --bican-user-network-name CAN \
+ncn# ./sls_updater_csm_1.3.py --sls-input-file sls_input_file.json --bican-user-network-name CAN \
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/expert_mode.md:90]: Script not renamed to match doc change

File: upgrade/1.3/scripts/sls/expert_mode.md [hunk 19]
```diff
@@ -152,7 +152,7 @@ As might be observed from the above example, due to pre-upgrade subnet sizes, or
 ### Expert Mode Process
 
 1. Review the existing networks and subnets in the SLS input file.
-2. Make an educated guess about how `sls_updater_csm_1.2.py` should work and run the program. A typical first pass is to `--preserve-existing-subnet-for-cmn external-dns`.
+2. Make an educated guess about how `sls_updater_csm_1.3.py` should work and run the program. A typical first pass is to `--preserve-existing-subnet-for-cmn external-dns`.
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/expert_mode.md:155]: Script not renamed to match doc change

File: upgrade/1.3/scripts/sls/expert_mode.md [hunk 20]
```diff
@@ -169,18 +169,18 @@ For this example of expert mode a very common use case is used with the followin
...
 2. The external-dns IP must be maintained to avoid operational changes to the site/customer network.
-3. The upgrade to CSM 1.2 will be a rolling upgrade, not a full outage. In order to prevent race conditions during the rolling upgrade with temporary IP overlaps, preservation of the K8S NCN IPs for CAN (CMN) are required.
+3. The upgrade to CSM 1.3 will be a rolling upgrade, not a full outage. In order to prevent race conditions during the rolling upgrade with temporary IP overlaps, preservation of the K8S NCN IPs for CAN (CMN) are required.
 
 The focus of the process that follows will be on the CMN IP allocations. The same process may be used for the CAN.
 
 **Process:**
 
-1. Review of the existing SLS file for CAN shows that NCN addresses are in the range: `10.103.11.2` to `10.103.11.14` with a DHCP Pool above this.
+1. Review of the existing SLS file for CAN shows that NCN addresses are in the range: `10.103.11.3` to `10.103.11.14` with a DHCP Pool above this.
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/expert_mode.md:178]:
```suggestion
1. Review of the existing SLS file for CAN shows that NCN addresses are in the range: `10.103.11.2` to `10.103.11.14` with a DHCP Pool above this.
```

File: upgrade/1.3/scripts/sls/expert_mode.md [hunk 21]
```diff
@@ -169,18 +169,18 @@ For this example of expert mode a very common use case is used with the followin
...
 
 2. An educated first pass is to run the updater while preserving the external-dns IP only and look for the CMN output:
 
     ```bash
-    ncn# ./sls_updater_csm_1.2.py \
+    ncn# ./sls_updater_csm_1.3.py \
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/expert_mode.md:183]: Script not renamed to match doc change

File: upgrade/1.3/scripts/sls/expert_mode.md [hunk 22]
```diff
@@ -218,7 +218,7 @@ The focus of the process that follows will be on the CMN IP allocations. The sam
 5. The next run of the upgrader for the example looks like:
 
     ```bash
-    ncn# ./sls_updater_csm_1.2.py \
+    ncn# ./sls_updater_csm_1.3.py \
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/sls/expert_mode.md:221]: Script not renamed to match doc change

File: upgrade/1.3/scripts/upgrade/prerequisites.sh [hunk 23]
```diff
@@ -675,8 +675,8 @@ if [[ $state_recorded == "0" && $(hostname) == "ncn-m001" ]]; then
 
     POD=$(kubectl -n services get pod -l app.kubernetes.io/instance=gitea -o json | jq -r '.items[] | .metadata.name')
     #
-    # Gitea change in 1.2 from /data to /var/lib/gitea, see which version we're
-    # backing up (in support of 1.2 -> 1.2 upgrades)
+    # Gitea change in 1.3 from /data to /var/lib/gitea, see which version we're
```
Comment (mharding-hpe) [inline upgrade/1.3/scripts/upgrade/prerequisites.sh:678]: This is not a change in 1.3

**Review by mharding-hpe — CHANGES_REQUESTED** [review]
Also, I thought we had decided to stop naming the directory in upgrade after the release. I can see it making sense to maybe split it up into different directories for different patch versions (like 1.0.0, 1.0.1, 1.0.10, etc), but anyone doing an upgrade with these docs will be going to 1.3, so having the 1.3 directory doesn't seem good (especially since it forces us to do this renaming each release).

**Comment by dzou-hpe** [comment]
closing this one and work on a new one after talking to @mharding-hpe

**Pull request closed without merging**